frontend/dockerfile/dockerfile2llb/convert_expose.go

```diff
@@ -0,0 +1,169 @@
+package dockerfile2llb
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/moby/buildkit/frontend/dockerfile/instructions"
+	"github.com/moby/buildkit/frontend/dockerfile/shell"
+	"github.com/pkg/errors"
+)
+
+func dispatchExpose(d *dispatchState, c *instructions.ExposeCommand, shlex *shell.Lex) error {
+	ports := []string{}
+	env := getEnv(d.state)
+	for _, p := range c.Ports {
+		ps, err := shlex.ProcessWords(p, env)
+		if err != nil {
+			return err
+		}
+		ports = append(ports, ps...)
+	}
+	c.Ports = ports
+
+	ps, err := parsePortSpecs(c.Ports)
+	if err != nil {
+		return err
+	}
+
+	if d.image.Config.ExposedPorts == nil {
+		d.image.Config.ExposedPorts = make(map[string]struct{})
+	}
+	for _, p := range ps {
+		d.image.Config.ExposedPorts[p] = struct{}{}
+	}
+
+	return commitToHistory(&d.image, fmt.Sprintf("EXPOSE %v", ps), false, nil, d.epoch)
+}
+
+// parsePortSpecs receives port specs in the format of [ip:]public:private/proto
+// and returns them as a list of "port/proto".
+func parsePortSpecs(ports []string) (exposedPorts []string, _ error) {
+	for _, p := range ports {
+		portProtos, err := parsePortSpec(p)
+		if err != nil {
+			return nil, err
+		}
+		exposedPorts = append(exposedPorts, portProtos...)
+	}
+	return exposedPorts, nil
+}
+
+// splitProtoPort splits a port(range) and protocol, formatted as "<portnum>/[<proto>]"
+// "<startport-endport>/[<proto>]". It returns an error if no port(range) or
+// an invalid proto is provided. If no protocol is provided, the default ("tcp")
+// protocol is returned.
+func splitProtoPort(rawPort string) (proto string, port string, _ error) {
+	port, proto, _ = strings.Cut(rawPort, "/")
+	if port == "" {
+		return "", "", errors.Errorf("no port specified: %s<empty>", rawPort)
+	}
+	proto = strings.ToLower(proto)
+	switch proto {
+	case "":
+		return "tcp", port, nil
+	case "tcp", "udp", "sctp":
```

Comment:
I think this is fine to keep `sctp` as long as host kernel and network stack support SCTP.